samza-core/src/main/scala/org/apache/samza/container/SamzaContainer.scala

```diff
@@ -462,6 +436,39 @@ object SamzaContainer extends Logging {
     else
       null
 
+    def maybeAsyncStreamTaskFactory : Object = {
+
+      val taskClassFactory = taskFactory match {
```

Comment:
I don't see much of the difference here. Anyways, it is scala coding style here and I don't see one way is better than the other. 
Also I have refactored the code a little bit more and seems match works better. Feel free to take a look at the revision and let me know if you still feel strong about this.